### PR TITLE
IT TN Fixes for #166

### PR DESCRIPTION
**Pull request by zoobereq**
# What does this PR do ?

A fix for [issue #166](https://github.com/NVIDIA/NeMo-text-processing/issues/166) reported for Italian TN.  Fixes the bug where the sentence-final period in sentences ending with domain is incorrectly normalized as part of the domain. The PR also includes support for social media tags, updated tests and a fix to `test_sparrowhawk_normalization.sh` which blocked Sparrowhawk testing.

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Have you signed your commits? Use ``git commit -s`` to sign.
- [x] Do all unittests finish successfully before sending PR?
   1) ``pytest`` or (if your machine does not have GPU) ``pytest --cpu`` from the root folder (given you marked your test cases accordingly `@pytest.mark.run_only_on('CPU')`).
   2) Sparrowhawk tests ``bash tools/text_processing_deployment/export_grammars.sh --MODE=test ...``
- [x] If you are adding a new feature: Have you added test cases for both `pytest` and Sparrowhawk [here](tests/nemo_text_processing).
- [x] Have you added ``__init__.py`` for every folder and subfolder, including `data` folder which has .TSV files?
- [ ] Have you followed codeQL results and removed unused variables and imports (report is at the bottom of the PR in github review box) ?
- [x] Have you added the correct license header `Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.` to all newly added Python files?
- [x] If you copied [nemo_text_processing/text_normalization/en/graph_utils.py](nemo_text_processing/text_normalization/en/graph_utils.py) your header's second line should be `Copyright 2015 and onwards Google, Inc.`. See an example [here](https://github.com/NVIDIA/NeMo-text-processing/blob/main/nemo_text_processing/text_normalization/en/graph_utils.py#L2).
- [x] Remove import guards (`try import: ... except: ...`) if not already done.
- [x] If you added a new language or a new feature please update the [NeMo documentation](https://github.com/NVIDIA/NeMo/blob/main/docs/source/nlp/text_normalization/wfst/wfst_text_normalization.rst) (lives in different repo).
- [x] Have you added your language support to [tools/text_processing_deployment/pynini_export.py](tools/text_processing_deployment/pynini_export.py).
  

**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation
- [x] Test

If you haven't finished some of the above items you can still open "Draft" PR.

### CODE REVIEW

File: tests/nemo_text_processing/it/data_text_normalization/test_cases_whitelist.txt [hunk 1]
```diff
@@ -1,5 +1,3 @@
-Avv.~avvocato
```
Comment (tbartley94) [inline tests/nemo_text_processing/it/data_text_normalization/test_cases_whitelist.txt:1]: Whhy deletion?

File: nemo_text_processing/text_normalization/it/verbalizers/electronic.py [hunk 2]
```diff
@@ -54,21 +55,24 @@ def add_space_after_char():
...
         user_name @= verbalize_characters
 
         convert_defaults = pynutil.add_weight(NEMO_NOT_QUOTE, weight=0.0001) | server_common | domain_common
         domain = convert_defaults + pynini.closure(insert_space + convert_defaults)
         domain @= verbalize_characters
 
-        domain = pynutil.delete("domain: \"") + domain + pynutil.delete("\"")
+        domain = pynutil.delete('domain: "') + domain + pynutil.delete('"')
         protocol = (
-            pynutil.delete("protocol: \"")
```
Comment (tbartley94) [inline nemo_text_processing/text_normalization/it/verbalizers/electronic.py:66]: just alias protocol as its own variable

File: nemo_text_processing/text_normalization/it/verbalizers/electronic.py [hunk 3]
```diff
@@ -54,21 +55,24 @@ def add_space_after_char():
...
 
-        user_name = pynutil.delete("username: \"") + add_space_after_char() + pynutil.delete("\"")
+        user_name = pynutil.delete('username: "') + add_space_after_char() + pynutil.delete('"')
         user_name @= verbalize_characters
 
         convert_defaults = pynutil.add_weight(NEMO_NOT_QUOTE, weight=0.0001) | server_common | domain_common
         domain = convert_defaults + pynini.closure(insert_space + convert_defaults)
         domain @= verbalize_characters
 
-        domain = pynutil.delete("domain: \"") + domain + pynutil.delete("\"")
```
Comment (tbartley94) [inline nemo_text_processing/text_normalization/it/verbalizers/electronic.py:64]: alias domain as its own variable

File: nemo_text_processing/text_normalization/it/verbalizers/electronic.py [hunk 4]
```diff
@@ -54,21 +55,24 @@ def add_space_after_char():
 
         verbalize_characters = pynini.cdrewrite(graph_symbols | graph_digit, "", "", NEMO_SIGMA)
 
-        user_name = pynutil.delete("username: \"") + add_space_after_char() + pynutil.delete("\"")
```
Comment (tbartley94) [inline nemo_text_processing/text_normalization/it/verbalizers/electronic.py:57]: alias username as its own variable

File: nemo_text_processing/text_normalization/it/taggers/electronic.py [hunk 5]
```diff
@@ -37,44 +36,41 @@ def __init__(self, deterministic: bool = True):
...
 
-        domain_common_graph = (
-            pynutil.insert("domain: \"")
-            + accepted_characters
-            + accepted_common_domains
-            + pynini.closure((accepted_symbols | dot) + pynini.closure(accepted_characters, 1), 0, 1)
-            + pynutil.insert("\"")
-        )
+        # email
+        username = pynutil.insert('username: "') + all_characters + pynutil.insert('"') + pynini.cross("@", " ")
```
Comment (tbartley94) [inline nemo_text_processing/text_normalization/it/taggers/electronic.py:53]: alias username as its own variable

File: nemo_text_processing/text_normalization/it/taggers/electronic.py [hunk 6]
```diff
@@ -37,44 +36,41 @@ def __init__(self, deterministic: bool = True):
...
 
+        # url
         protocol_start = pynini.accep("https://") | pynini.accep("http://")
         protocol_end = (
             pynini.accep("www.") if deterministic else pynini.accep("www.") | pynini.cross("www.", "vu vu vu.")
         )
         protocol = protocol_start | protocol_end | (protocol_start + protocol_end)
-        protocol = pynutil.insert("protocol: \"") + protocol + pynutil.insert("\"")
-        graph |= protocol + insert_space + (domain_graph | domain_common_graph)
+        protocol = pynutil.insert('protocol: "') + protocol + pynutil.insert('"')
```
Comment (tbartley94) [inline nemo_text_processing/text_normalization/it/taggers/electronic.py:70]: alias protocol as its own variab.e

**Review by tbartley94 — CHANGES_REQUESTED** [review]

File: tests/nemo_text_processing/it/data_text_normalization/test_cases_whitelist.txt (hunk 1 above)
Comment (zoobereq) [reply tests/nemo_text_processing/it/data_text_normalization/test_cases_whitelist.txt:1]: The case passes `pyTests` but fails with `Sparrowhawk`.

File: nemo_text_processing/text_normalization/it/verbalizers/electronic.py (hunk 2 above)
Comment (zoobereq) [reply nemo_text_processing/text_normalization/it/verbalizers/electronic.py:66]: Done.

File: nemo_text_processing/text_normalization/it/verbalizers/electronic.py (hunk 3 above)
Comment (zoobereq) [reply nemo_text_processing/text_normalization/it/verbalizers/electronic.py:64]: Done.

File: nemo_text_processing/text_normalization/it/verbalizers/electronic.py (hunk 4 above)
Comment (zoobereq) [reply nemo_text_processing/text_normalization/it/verbalizers/electronic.py:57]: Done.

File: nemo_text_processing/text_normalization/it/taggers/electronic.py (hunk 6 above)
Comment (zoobereq) [reply nemo_text_processing/text_normalization/it/taggers/electronic.py:70]: Done.

File: nemo_text_processing/text_normalization/it/taggers/electronic.py (hunk 5 above)
Comment (zoobereq) [reply nemo_text_processing/text_normalization/it/taggers/electronic.py:53]: Done.

**Comment by ekmb** [comment]
@tbartley94 , @zoobereq this PR changes en/graph - CI folder for Eng should've been updated 
Eng SH tests are failing afeter this commit https://github.com/NVIDIA/NeMo-text-processing/commits/main/

**Comment by tbartley94** [comment]
@ekmb How do I open a revert with a signature sign off? Trying to revert the branch but it's triggering Dco failures.

**Pull request merged**